### PR TITLE
Adding food to recipe - User flow improvements

### PR DESCRIPTION
**Pull request by Cwighty**
When creating a recipe you can now add foods with different amount units is no longer restricted to the unit type of the food serving size. If you add an amount that is from a different unit category (Grams -> Qty), the user will be asked to enter a sample conversion. This is stored and used to convert food nutrient amounts into recipe nutrient amounts.

If they ever add that same food in the same unit amount, they won't be asked again, but it will display what they entered before.
To help them answer the question, there is a link to google for the amount.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN dnf -y install \
       tar \
       gzip \
       git \
-      dotnet-sdk-8.0 \
```
Comment (garv5014) [inline .devcontainer/Dockerfile:8]: Did this help with the vs code stuff by chance or nah?

File: nutrinova-api/NutrinovaData.UnitTests/Features/Recipes/RecipeFoodTotalerSteps.cs [hunk 2]
```diff
@@ -69,14 +69,16 @@ public void GivenTheFollowingFoodNutrients(Table table)
...
 
     foreach (var foodNutrient in foodNutrients)
     {
       foodNutrient.Foodplan = foods?.FirstOrDefault(f => f.Id == foodNutrient.FoodplanId) ?? new FoodPlan { Id = foodNutrient.FoodplanId };
       foodNutrient.Nutrient = nutrients?.FirstOrDefault(n => n.Id == foodNutrient.NutrientId) ?? new Nutrient { Id = foodNutrient.NutrientId };
+      foodNutrient.Unit = units?.FirstOrDefault(u => u.Id == foodNutrient.UnitId) ?? new Unit { Id = foodNutrient.UnitId };
     }
 
-    foreach (var food in foods)
+    foreach (var food in foods!)
```
Comment (garv5014) [inline nutrinova-api/NutrinovaData.UnitTests/Features/Recipes/RecipeFoodTotalerSteps.cs:81]: What does that ! Do?

**Comment by garv5014** [comment]
Lgtm

**Review by garv5014 — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (Cwighty) [reply .devcontainer/Dockerfile:8]: No I've just been editing the backend in visual studio

File: nutrinova-api/NutrinovaData.UnitTests/Features/Recipes/RecipeFoodTotalerSteps.cs (hunk 2 above)
Comment (Cwighty) [reply nutrinova-api/NutrinovaData.UnitTests/Features/Recipes/RecipeFoodTotalerSteps.cs:81]: A useless null check, I should probably take it out

**Pull request merged**